prov/tcp/src/tcpx_progress.c

```diff
@@ -249,14 +308,26 @@ static int tcpx_get_rx_pe_entry(struct tcpx_rx_detect *rx_detect,
 			return ret;
 
 		break;
-	case ofi_op_read_req:
 	case ofi_op_read_rsp:
-	case ofi_op_write_rsp:
-		/* todo  complete these cases for rma*/
+		if (dlist_empty(&tcpx_ep->rma_list.list))
+			return -FI_EINVAL;
+
+		entry = dlist_find_first_match(&tcpx_ep->rma_list.list,
+					       tcpx_match_read_rsp,
+					       rx_detect);
+		if (!entry)
+			return -FI_EINVAL;
```

Comment:
Any errors in this path are unexpected protocol errors.  Please add log messages to indicate a possibly fatal issue has occurred.